src/betterproto/__init__.py

```diff
@@ -169,7 +169,17 @@ class Casing(builtin_enum.Enum):
     SNAKE = snake_case  #: A snake_case sterilization function.
 
 
-PLACEHOLDER: Any = object()
+class Placeholder:
+    def __copy__(self):
+        return self
+
+    def __deepcopy__(self, _):
+        return self
+
+
+# We can't simply use object() here because pydantic automatically performs deep-copy of mutable default values
+# See https://github.com/danielgtaylor/python-betterproto/issues/606
+PLACEHOLDER: Any = Placeholder()
```

Comment:
```suggestion
class Placeholder:
    __slots__ = ()

    def __repr__(self) -> str:
        return "<PLACEHOLDER>"

    def __copy__(self) -> Self:
        return self

    def __deepcopy__(self, _) -> Self:
        return self


# We can't simply use object() here because pydantic automatically performs deep-copy of mutable default values
# See #606
PLACEHOLDER: Any = Placeholder()
```

Reply:
The changes look fine, thanks!